### PR TITLE
feat: support aggregate search

### PR DESCRIPTION
**Pull request by PainterPuppets**
reslove: https://github.com/Magickbase/ckb-explorer-public-issues/issues/519

### CODE REVIEW

File: src/services/ExplorerService/fetcher.ts [hunk 1]
```diff
@@ -293,6 +314,14 @@ export const apiFetcher = {
     // TODO: When will it return an empty result?
     v1GetNullableWrapped<{ data: string }>(`/cell_output_data/${id}`).then(res => res?.attributes.data ?? null),
 
+  fetchAggregateSearchResult: (param: string) =>
+    v1Get<AggregateSearchResult[]>('suggest_queries', {
+      params: {
+        q: param,
+        filter_by: 0,
```
[the remark this answers is not in the document]
Comment (PainterPuppets) [reply src/services/ExplorerService/fetcher.ts:321]: The api provided by the backend, the code is here: https://github.com/nervosnetwork/ckb-explorer/pull/1790/files#diff-240d8fa62d7d98e879be4a2dc4a31b67265baee47f7f419a67226a483c71ebd7R10

[the remark this answers is not in the document]
Comment (Keith-CY) [reply src/services/ExplorerService/fetcher.ts:321]: It can be defined as an enum as
```
0 for aggregate_query
1 for single_query
```
to make it clear

**Review by Keith-CY — APPROVED** [review]

**Comment by PainterPuppets** [comment]
supported token collection result display

image

**Comment by FrederLu** [comment]
PRD  https://vs0cjf.axshare.com/?id=908ssd&g=1&p=%E6%90%9C%E7%B4%A2%E4%BC%98%E5%8C%96%E8%AF%B4%E6%98%8E&sc=3

1、When the number of blocks being searched is greater than the current number of blocks, clicking Enter will jump to the block countdown page.
image
image


---
2、When searching at the top, Enter or clicking the search result will not jump to the corresponding page.
image

---
3、`Lock script`(0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8) and `Type script`(0x8536c9d5d908bd89fc70099e4284870708b6632356aad98734fcf43f6f71c304)  is not Chinese when the page displays Chinese.
image

image
image
When using type script at the same time, the search results display `text`?

---
4、Use btc address but no search results.
Btc address: tb1qwq82nch2wc0jtymavfv2cxsju9trghn0ykhqx8
https://ckb-explorer-frontend-in-magickbase-repo-8dt95odvy-magickbase.vercel.app/address/tb1qwq82nch2wc0jtymavfv2cxsju9trghn0ykhqx8
image

---
5、Use btc tx but no search results.
BTC Tx :4ea970151577aff9f134bd50ce666d304c979f8d204c4fa8f4cd0a1bf7c3bbd1
https://ckb-explorer-frontend-in-magickbase-repo-8dt95odvy-magickbase.vercel.app/transaction/0x769c208860f364c602afb89c11887c7d8a8630bebeeb4d6ac2c65b6a3362d760
image

**Comment by Keith-CY** [comment]
Any update on this feature @PainterPuppets

**Comment by PainterPuppets** [comment]
supported search result category filter & did / btc address search
image
image

**Comment by PainterPuppets** [comment]
> PRD https://vs0cjf.axshare.com/?id=908ssd&g=1&p=%E6%90%9C%E7%B4%A2%E4%BC%98%E5%8C%96%E8%AF%B4%E6%98%8E&sc=3
> 
> UI https://www.figma.com/file/6XNoimRDbFTTNm016rbIdU/Magickbase?type=design&node-id=35067-23245&mode=design&t=f9reojugIfk2nzmp-0
> 
> 1、When the number of blocks being searched is greater than the current number of blocks, clicking Enter will jump to the block countdown page. image image
> 
> 2、When searching at the top, Enter or clicking the search result will not jump to the corresponding page. image
> 
> 3、`Lock script`(0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8) and `Type script`(0x8536c9d5d908bd89fc70099e4284870708b6632356aad98734fcf43f6f71c304) is not Chinese when the page displays Chinese. image
> 
> image image When using type script at the same time, the search results display `text`?
> 4、Use btc address but no search results. Btc address: tb1qwq82nch2wc0jtymavfv2cxsju9trghn0ykhqx8 https://ckb-explorer-frontend-in-magickbase-repo-8dt95odvy-magickbase.vercel.app/address/tb1qwq82nch2wc0jtymavfv2cxsju9trghn0ykhqx8 image
> 
> 5、Use btc tx but no search results. BTC Tx :4ea970151577aff9f134bd50ce666d304c979f8d204c4fa8f4cd0a1bf7c3bbd1 https://ckb-explorer-frontend-in-magickbase-repo-8dt95odvy-magickbase.vercel.app/transaction/0x769c208860f364c602afb89c11887c7d8a8630bebeeb4d6ac2c65b6a3362d760 image
> 
> 6、Use .bit but no search results. https://pudge.explorer.nervos.org/address/ckt1qzfhdsa4syv599s2s3nfrctwga70g0tu07n9gpnun9ydlngf5vsnwqg9k22096uxuljzg6kjsdcmacxtm2m4vw8yqkeffuhtsmn7gfr262phr0hqe0dtw43cushnmkvy image
> 
> 7、The data style in the search results is abnormal. Refer to the visible part of data before and after the UI style, using `...` in the middle.
> 
> UI image image image image
> 
> 8、Like typing `aaaa`, the page lacks multiple data tabs, and the search box lacks filtering options.
> 
> image
> UI: image
> 
> image

1. We don't seem to have a block countdown page at the moment, so i talked to the @Sven-TBD , then PM said we'll ignore the logic for now.
2. I cant reproducing it. do you have a video?
3. fixed
4. supported
5. There doesn't seem to be a way for the front-end to know if a btc transaction is rgb++ or not, and there doesn't seem to be a way to correspond to a ckb transaction, so it may require back-end support for searching btc transactions @rabbitz 
6. supported
7. supported
8. Added filter for search result panel, I felt that the filtering in front of the search box was a bit of a duplicate of the filtering on the search results panel, so I discussed it with the @Sven-TBD and concluded that it could be left off for now

**Comment by FrederLu** [comment]
>2. I cant reproducing it. do you have a video?

The operation jumped after this submission.

>5. There doesn't seem to be a way for the front-end to know if a btc transaction is rgb++ or not, and there doesn't seem to be a way to correspond to a ckb transaction, so it may require back-end support for searching btc transactions @rabbitz

 @rabbitz You need to take a look at this.

---
9、The search result style of Type script(0x8536c9d5d908bd89fc70099e4284870708b6632356aad98734fcf43f6f71c304) needs to be the same as Lock script(0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a 8637b17723bbda3cce8) Be consistent.

image
image

---
10、The spacing between selections needs to be adjusted according to the UI, and is currently too small.At the same time, the selected state here only requires a single selection. Multiple selections are now possible.
image

**Comment by rabbitz** [comment]
image

Need to add `0x` in front of btc txid @PainterPuppets

**Comment by PainterPuppets** [comment]
> 0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a 8637b17723bbda3cce8

fixed

**Comment by PainterPuppets** [comment]
> > 2. I cant reproducing it. do you have a video?
> 
> The operation jumped after this submission.
> 
> > 5. There doesn't seem to be a way for the front-end to know if a btc transaction is rgb++ or not, and there doesn't seem to be a way to correspond to a ckb transaction, so it may require back-end support for searching btc transactions @rabbitz
> 
> @rabbitz You need to take a look at this.
> 
> 9、The search result style of Type script(0x8536c9d5d908bd89fc70099e4284870708b6632356aad98734fcf43f6f71c304) needs to be the same as Lock script(0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a 8637b17723bbda3cce8) Be consistent.
> 
> image image
> 10、The spacing between selections needs to be adjusted according to the UI, and is currently too small.At the same time, the selected state here only requires a single selection. Multiple selections are now possible. image
> 
> 11、Entering `aa` according to the search logic requires searching for `Token Collection` data. image image At the same time, the data `aaaaa` should also be displayed when `aa` or `aaa` is entered. image
> 
> 12、The `Token Collection` data displayed on the page does not match the api. image image
> 
> 13、When searching using `BTC Tx:4ea970151577aff9f134bd50ce666d304c979f8d204c4fa8f4cd0a1bf7c3bbd1`, refer to the UI to improve the search result style. There is also the `RGB++` tag. CKB transaction ID (above) - corresponding api ckb_transaction_hash; BTC TXID (below)--corresponding api transaction_hash; image image

fixed

**Comment by FrederLu** [comment]
image

Due to adjustments to the `btc address`-`tb1qkk0373ap96786j5532n3a8agxnn657d9j7w206` search API, the current page display is inconsistent with the API data and needs to be adjusted.

**Comment by PainterPuppets** [comment]
> tb1qkk0373ap96786j5532n3a8agxnn657d9j7w206

fixed

**Comment by PainterPuppets** [comment]
> image
> Due to adjustments to the `btc address`-`tb1qkk0373ap96786j5532n3a8agxnn657d9j7w206` search API, the current page display is inconsistent with the API data and needs to be adjusted.

fixed

**Comment by FrederLu** [comment]
>image

When multiple options appear, the problem of only single selection still exists.

---
image
image
14、The search results show a lack of reasonable adaptation. When using `...`, you need to give priority to the later display positions.

**Comment by PainterPuppets** [comment]
image

my mistake, I thought this was a multiple checkbox.

had fixed

**Comment by PainterPuppets** [comment]
> > image
> 
> When multiple options appear, the problem of only single selection still exists.
> 
> image image
> 14、The search results show a lack of reasonable adaptation. When using `...`, you need to give priority to the later display positions.
> 
> Reference style image

my mistake, I thought this was a multiple checkbox.

had fixed & optimize some search result item ui

image
image

**Comment by FrederLu** [comment]
image
To avoid ambiguity, it can be written as block height data.

**Review by FrederLu — APPROVED** [review]

**Pull request merged**